common/stash/src/main/java/com/bazaarvoice/emodb/common/stash/StashScanIterator.java

```diff
@@ -43,27 +44,32 @@ private void moveToNextSplit() {
     private void closeCurrentSplit() {
         if (_currentIterator != null) {
             try {
-                Closeables.close(_currentIterator, true);
-            } catch (IOException e) {
-                // Already logged and caught
+                _currentIterator.close();
+            } catch (IOException ignore) {
+                // ignored
             }
             _currentIterator = null;
         }
     }
 
     @Override
-    protected Map<String, Object> computeNext() {
-        while (_currentIterator != null) {
-            // Return the next row from the current split if available
-            if (_currentIterator.hasNext()) {
-                return _currentIterator.next();
-            }
+    protected Spliterator<Map<String, Object>> getSpliterator() {
```

Comment:
This is really minor, but this pattern is really common.  Pretty much every where `SpiteratorIterator` is used the `getSpliterator()` method returns an implementation of `AbstractSpliterator`.  Not that this needs to change, but there may be a convenience factor of having a superclass which provides all of that plumbing for you.